### ISSUE TITLE
Add typing

### ISSUE THREAD

**Issue by merwok** [caravancoop/configstore#25]
Add type declarations (I think Python 3 native syntax, and drop 2.7 support)

Run mypy from tox

Run mypy in CI

**Comment by merwok** [comment]
@just1602 intéressé?

**Comment by just1602** [comment]
We should think about contributing typing stub file for boto3 and dotenv library.

**Comment by merwok** [comment]
Good idea!  The first goal here was to make sure the store and backend classes cooperate in the right way, in complement to integration tests.  Contributing a stub file to django-dotenv could be interesting!

**Comment by merwok** [comment]
@just1602 Any feedback on mypy vs pyre?

**Comment by just1602** [comment]
I didn't have the time to test yesterday. I'll have to test with `ABC` to see if it works with mypy. Otherwise we'll have to stay with `pyre-check` or open issue to `mypy` repo.

**Comment by just1602** [comment]
@merwok because `pyre` handle the inheritance correctly and not `mypy`. Mypy give me error that the parameter `self` is missing when I call a method of an object.

**Comment by merwok** [comment]
That seems like a pretty fundamental Python thing to handle.
https://mypy.readthedocs.io/en/latest/cheat_sheet_py3.html#classes

**Comment by merwok** [comment]
I tried running mypy

> configstore/store.py:23: error: Too few arguments for "get_setting" of "Backend"
> configstore/store.py:23: error: Argument 1 to "get_setting" of "Backend" has incompatible type "str"; expected "Backend"

I think the problem is that the type inference things that the objects in self._backends are Backend classes, not instances.

**Comment by just1602** [comment]
But pyre handle this gracefully. Maybe we should open an issue to see what's going on.

**Comment by merwok** [comment]
Validates if I use `Tuple[Backend]` without `Type` (after quickly checking the doc for Type that seems to be useful for more complicated setups).
